### PR TITLE
API/DEPR: replace "raise_conflict" with "errors" for df.update

### LINKED ISSUE #23585: API/DEPR: replace raise_conflict-kwarg in df.update with errors
From [review](https://github.com/pandas-dev/pandas/pull/23192/files#r225758408) in #23192:
> pls create [an issue] to deprecate this arg and rename to `errors=`

The idea is to replace the `raise_conflict=True|False` with `errors='ignore'|'raise'`.

Same goes for `Panel`.

### PR DESCRIPTION
**Pull request by h-vetinari**
- [x] closes #23585 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

This is split off from #23192, as requested per review (because the PR was getting larger and larger).

### CODE REVIEW

**Review by gfyoung — APPROVED** [review]
cc @jreback 
cc @datapythonista (b/c of the doc changes)

File: pandas/core/frame.py [hunk 1]
```diff
@@ -5255,17 +5257,21 @@ def update(self, other, join='left', overwrite=True, filter_func=None,
...
               the original DataFrame.
 
-        filter_func : callable(1d-array) -> boolean 1d-array, optional
+        filter_func : callable(1d-array) -> bool 1d-array, optional
             Can choose to replace values other than NA. Return True for values
             that should be updated.
-        raise_conflict : bool, default False
-            If True, will raise a ValueError if the DataFrame and `other`
+        errors : {'raise', 'ignore'}, default 'ignore'
+            If 'raise', will raise a ValueError if the DataFrame and `other`
```
Comment (jreback) [inline pandas/core/frame.py]: versionchanged tag

File: pandas/core/frame.py [hunk 2]
```diff
@@ -5255,17 +5257,21 @@ def update(self, other, join='left', overwrite=True, filter_func=None,
...
             both contain non-NA data in the same place.
 
+        Returns
+        -------
+        None : method directly changes calling object
+
         Raises
         ------
```
Comment (jreback) [inline pandas/core/frame.py]: can you add an if for valid errors values (e.g. raise/error), and test for passing bad values in the tests themselves

File: pandas/core/panel.py [hunk 3]
```diff
@@ -1377,25 +1382,33 @@ def join(self, other, how='left', lsuffix='', rsuffix=''):
...
+                     mapping={False: 'ignore', True: 'raise'})
     def update(self, other, join='left', overwrite=True, filter_func=None,
-               raise_conflict=False):
+               errors='ignore'):
         """
-        Modify Panel in place using non-NA values from passed
-        Panel, or object coercible to Panel. Aligns on items
+        Modify Panel in place using non-NA values from other Panel.
+
+        May also use object coercible to Panel. Will align on items.
```
[the remark this answers is not in the document]
Comment (jreback) [reply pandas/core/panel.py]: this is fine

File: pandas/tests/frame/test_combine_concat.py [hunk 4]
```diff
@@ -322,7 +322,9 @@ def test_update_raise(self):
         other = DataFrame([[2., nan],
                            [nan, 7]], index=[1, 3], columns=[1, 2])
         with pytest.raises(ValueError, match="Data overlaps"):
-            df.update(other, raise_conflict=True)
+            df.update(other, errors='raise')
```
Comment (jreback) [inline pandas/tests/frame/test_combine_concat.py]: can you move this to another test, e.g. test_update_deprecation

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**